Model/Observer/OrderSaveObserver.php

```diff
@@ -0,0 +1,52 @@
+<?php
+namespace Wizzy\Search\Model\Observer;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Wizzy\Search\Services\Indexer\IndexerManager;
+use Magento\ConfigurableProduct\Model\Product\Type\Configurable;
+
+class OrderSaveObserver implements ObserverInterface
+{
+    private $indexer;
+    private $configurable;
+
+    public function __construct(
+        IndexerManager $indexerManager,
+        Configurable $configurable
+    ) {
+        $this->indexer = $indexerManager->getProductsIndexer();
+        $this->configurable = $configurable;
+    }
+    public function execute(Observer $observer)
+    {
+        $order = $observer->getEvent()->getOrder();
+        if ($order->getState() == 'complete' || $order->getState() == 'closed') {
+            $orderItems = $order->getAllItems();
```

Comment:
Create a base class called BaseOrdersObserver to Execute the code from Line 25 to Line 32 . create a function called `addOrderProductsInSync($order)` can use it in this class as well as above class.